chia/rpc/wallet_rpc_api.py

```diff
@@ -816,6 +817,12 @@ async def cc_spend(self, request):
         wallet: CCWallet = self.service.wallet_state_manager.wallets[wallet_id]
         puzzle_hash: bytes32 = decode_puzzle_hash(request["inner_address"])
 
+        memo: Optional[bytes] = None
+        if "memo" in request:
+            memo = hexstr_to_bytes(request["memo"])
+            if len(memo) > 1000:
```

Comment:
in the full node we only accept this field to be 32 bytes or shorter. That's an arbitrary limit we set though, and it doesn't affect consensu.

Reply:
It's useful to have a longer length. Is there an attack vector here? Maybe we can limit it at 1024 like we do for announcements?

Reply:
@arvidn when you say "we only accept this field", do you mean we only _index_ the field if it's 32 bytes? OR will the mempool reject a CREATE_COIN that has a memo longer than 32 bytes?

Reply:
I mean the *parser* will ignore it if it's greater than 32 bytes. Nothing is rejected (as that would be a consensus fork). So, as it is right now, the rust parser won't pass it on if it's longer. There's no problem making the parser accept longer strings, but then we would probably need to restrict the indexer to ignore coins with long keys.

Reply:
I don't think there's an attack per-se, but an index in the database with 1000 byte keys wouldn't be very fast. It's not mandatory to build this index, but our full node will do it unconditionally (at least for now).

Reply:
I see, I think it would make more sense to restrict the indexer so that you can extract longer memos easily if you want to, but it doesn't clog the databases of nodes that don't care.

Reply:
I would just leave out this check. Returning a response with no matches is sufficient. It seems like a weird place to enforce this restriction since it doesn't actually prevent any DOS or anything.

Reply:
OK, we can leave it out. There is still an upper limit, if you create a TX with higher cost than the block allows, it will fail to go into the mempool.